lib/logger.ts

```diff
@@ -1,276 +1,19 @@
-   */
-  verboseLogs: string[];
-}
-
-/**
- * Get a Logger that will store its logs in memory.
- */
-export function getInMemoryLogger(options: LoggerOptions = {}): InMemoryLogger {
-  const allLogs: string[] = [];
-  const infoLogs: string[] = [];
-  const errorLogs: string[] = [];
-  const warningLogs: string[] = [];
-  const sectionLogs: string[] = [];
-  const verboseLogs: string[] = [];
-  return {
-    allLogs,
-    infoLogs,
-    errorLogs,
-    warningLogs,
-    sectionLogs,
-    verboseLogs,
-    logInfo: getLogFunction(options.logInfo, (text: string) => {
-      allLogs.push(text);
-      infoLogs.push(text);
-      return Promise.resolve();
-    }),
-    logError: getLogFunction(options.logError, (text: string) => {
-      allLogs.push(text);
-      errorLogs.push(text);
-      return Promise.resolve();
-    }),
-    logWarning: getLogFunction(options.logWarning, (text: string) => {
-      allLogs.push(text);
-      warningLogs.push(text);
-      return Promise.resolve();
-    }),
-    logSection: getLogFunction(options.logSection, (text: string) => {
-      allLogs.push(text);
-      sectionLogs.push(text);
-      return Promise.resolve();
-    }),
-    logVerbose: getLogFunction(options.logVerbose, (text: string) => {
-      allLogs.push(text);
-      verboseLogs.push(text);
-      return Promise.resolve();
-    }, false)
-  };
-}
-
-export interface AzureDevOpsLoggerOptions extends LoggerOptions {
-  toWrap?: Logger;
-}
-
-/**
- * Get a Logger that will output prefixes for certain types of logs in the Azure DevOps environment.
- */
-export function getAzureDevOpsLogger(options: AzureDevOpsLoggerOptions = {}): Logger {
-  const innerLogger: Logger = options.toWrap || getConsoleLogger({
-    ...options,
-    logError: ("logError" in options ? options.logError : (text: string) => Promise.resolve(console.log(text))),
-  });
-  return wrapLogger(innerLogger, {
-    logError: (text: string) => innerLogger.logError(`##[error]${text}`),
-    logInfo: true,
-    logSection: (text: string) => innerLogger.logSection(`##[section]${text}`),
-    logWarning: (text: string) => innerLogger.logWarning(`##[warning]${text}`),
-    logVerbose: true
-  });
-}
+import { getBooleanArgument } from "./commandLine";
+import { getConsoleLogger, getAzureDevOpsLogger, LoggerOptions, Logger } from "@azure/logger-js";
 
 /**
  * Get the default Logger based on the command line arguments.
  * @returns The default Logger based on the command line arguments.
  */
 export function getDefaultLogger(options: LoggerOptions = {}): Logger {
-  if (options.logVerbose == undefined) {
-    options.logVerbose = getBooleanArgument("verbose");
-  }
-  return getBooleanArgument("azure-devops") ? getAzureDevOpsLogger(options) : getConsoleLogger(options);
+    if (options.logVerbose == undefined) {
+        options.logVerbose = getBooleanArgument("verbose");
+    }
+    return getBooleanArgument("azure-devops") ? getAzureDevOpsLogger(options) : getConsoleLogger(options);
 }
```

Comment:
I definitely recommend moving this to `@azure/logger-js` too.

Reply:
The reason why I haven't moved that is the `getBooleanArgument` which takes dependency on commandLine file in this package. What's the good solution for avoiding circular dependency? Implement the command line arguments in the package or extract command line stuff to an external package?

Reply:
Ah, good point. I'd recommend copying this function to `@azure/logger-js` but add an optional boolean property to `LoggerOptions` named `azureDevOps`. Then change this function to reference the `getDefaultLogger()` function exported from `@azure/logger-js` and the command line argument logic remains here to avoid the circular dependency. Does that make sense?

Reply:
Take a look if that's what you meant.